cache/test_cache.py

```diff
@@ -810,6 +810,126 @@ def test_cache_POST_request_not_cached(self, name, extra_header):
         self.assertNotIn("age", client.last_response.headers.keys())
         self.assertEqual(len(server.requests), 2)
 
+    @parameterize.expand(
+        [
+            param(
+                name="GET_POST",
+                first_method="GET",
+                second_method="POST",
+                second_headers=[],
+                sleep_interval=0,
+            ),
+            param(
+                name="GET_GET",
+                first_method="GET",
+                second_method="GET",
+                second_headers=[("cache-control", "max-age=1")],
+                sleep_interval=2,
+            ),
+            param(
+                name="GET_HEAD",
+                first_method="GET",
+                second_method="HEAD",
+                second_headers=[("cache-control", "max-age=1")],
+                sleep_interval=2,
+            ),
+            param(
+                name="POST_POST",
+                first_method="POST",
+                second_method="POST",
+                second_headers=[],
+                sleep_interval=0,
+            ),
+            param(
+                name="POST_GET",
+                first_method="POST",
+                second_method="GET",
+                second_headers=[("cache-control", "max-age=1")],
+                sleep_interval=2,
+            ),
+            param(
+                name="POST_HEAD",
+                first_method="POST",
+                second_method="HEAD",
+                second_headers=[("cache-control", "max-age=1")],
+                sleep_interval=2,
+            ),
+        ]
+    )
+    def test_several_entries(
+        self, name, first_method, second_method, second_headers, sleep_interval
+    ):
+        server = self.get_server("deproxy")
+        client = self.get_client("deproxy")
+
+        self.start_all_services()
+        if second_method == "HEAD":
+            self.disable_deproxy_auto_parser()
+
+        server.set_response(
+            "HTTP/1.1 200 OK\r\n"
+            + "Connection: keep-alive\r\n"
+            + "Content-Length: 11\r\n"
+            + "Content-Location: /index.html\r\n"
+            + "Cache-Control: public, s-maxage=5\r\n"
+            + "\r\n"
+            + "First body."
+        )
+
+        client.send_request(
+            client.create_request(method=first_method, uri="/index.html", headers=[]),
+            expected_status_code="200",
+        )
+        self.assertNotIn("age", client.last_response.headers.keys())
+        self.assertEqual(len(server.requests), 1)
+        self.assertEqual("First body.", client.last_response.body)
+
+        time.sleep(sleep_interval)
```

Comment:
These tests should not check `Cache-Control` header. We have separate files for that - `cache/test_cache_control.py` and `cache/test_h2_cache_control.py`. This class contain tests for different methods with cache. So I think we should not use `time.sleep` here. Probably I am wrong and the description in the test will help me understand better.